tensorrt_llm/_torch/pyexecutor/sampler.py

```
@@ -449,19 +492,42 @@ def get_generator(self, device: torch.device) -> torch.Generator:
             self._generator.manual_seed(self._global_seed)
         return self._generator
 
-    def handle_logprobs(self, request: LlmRequest, state: SampleStateTorch, *,
+    def _maybe_get_top_logprobs(self, request: LlmRequest,
+                                state: SampleStateTorch, *, beam: int,
+                                count: int):
+        current_slice = slice(0, count), request.py_seq_slot, beam
+        if request.calculate_top_logprobs:
+            assert state.host.log_probs is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)